libnopegl/src/utils.h

```diff
@@ -127,6 +127,19 @@
 
 #define NGLI_HAS_ALL_FLAGS(a, b) (((a) & (b)) == (b))
 
+typedef void (*ngli_freep_func)(void **sp);
+
+struct ngli_rc {
+    size_t count;
+    ngli_freep_func freep;
+};
+
+#define NGLI_RC_REF(s) ngli_rc_ref((struct ngli_rc *)(s))
+#define NGLI_RC_UNREFP(sp) ngli_rc_unrefp((struct ngli_rc **)(sp))
```

Comment:
I feel like it makes sense to have `NGLI_RC_NEW(fn) (struct ngli_rc) { .count=1, .freep=fn }` to simplify the next commit

Reply:
Done. NGLI_RC_CREATE().